### ISSUE TITLE
False positive on [no-role-prefix]

### ISSUE THREAD

**Issue by katelovescode** [ansible/ansible-lint#4925]
##### Summary

when running ansible-lint on this file in `roles/proxmox/tasks/main.yml` the linter throws an error telling me that all variables should be prefixed by `proxmox_`, but they are.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version

ansible-lint 26.1.1 using ansible-core:2.20.1 ansible-compat:25.12.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```

- ansible installation method: In a `pyenv-virtualenv` environment, `pipx install ansible`
- ansible-lint installation method: In a `pyenv-virtualenv` environment: `pipx inject --include-apps ansible ansible-lint`

##### STEPS TO REPRODUCE

```console (paste below)
# playbook.yml
- name: Proxmox node, cluster and VM management
  hosts: enterprise
  roles:
    - proxmox

# roles/proxmox/tasks/main.yml
---
- name: Create admin user
  ansible.builtin.import_tasks:
    file: admin_user.yml

# roles/proxmox/tasks/admin_user.yml
---
- name: Create linux user for admin
  become: true
  ansible.builtin.user:
    name: "{{ proxmox_admin_username }}"
    password: "{{ proxmox_admin_hashed_password }}"
    group: sudo
    shell: /bin/bash
  register: linux_user_step
- name: Add SSH key for admin user
  become: true
  ansible.posix.authorized_key:
    user: "{{ proxmox_admin_username }}"
    key: "{{ proxmox_ansible_public_key }}"
- name: Configure admin user in Proxmox
  community.proxmox.proxmox_user:
    api_host: "{{ proxmox_ansible_host }}"
    api_user: "root@pam"
    api_password: "{{ proxmox_root_password }}"
    comment: Managed by Ansible
    email: "{{ proxmox_admin_email }}"
    firstname: "{{ proxmox_admin_firstname }}"
    enable: true
    userid: "{{ proxmox_admin_api_user }}"
- name: Add admin permissions
  community.proxmox.proxmox_access_acl:
    api_host: "{{ proxmox_ansible_host }}"
    api_user: "root@pam"
    api_password: "{{ proxmox_root_password }}"
    path: /
    type: user
    state: present
    ugid: "{{ proxmox_admin_api_user }}"
    roleid: Administrator
    propagate: true
```

##### Desired Behavior

I would expect this to pass linting, as all vars are prefixed with `proxmox_`

##### Actual Behavior

```paste below
ansible-lint -vvv playbook.yml

DEBUG    Examining roles/proxmox/tasks/admin_user.yml of type tasks
[DEPRECATION WARNING]: Importing 'to_native' from 'ansible.module_utils._text' is deprecated. This feature will be removed from ansible-core version 2.24. Use ansible.module_utils.common.text.converters instead.
DEBUG    Examining roles/proxmox/tasks/install_templates.yml of type tasks
DEBUG    Examining roles/ansible_user/tasks/main.yml of type tasks
WARNING  /Users/kate/.local/pipx/venvs/ansible/lib/python3.13/site-packages/ansible/utils/display.py:719 AnsibleWarning Deprecation warnings can be disabled by setting `deprecation_warnings=False` in ansible.cfg.
WARNING  /Users/kate/.local/pipx/venvs/ansible/lib/python3.13/site-packages/ansible/utils/display.py:719 AnsibleWarning Deprecation warnings can be disabled by setting `deprecation_warnings=False` in ansible.cfg.
WARNING  Listing 1 violation(s) that are fatal
var-naming[no-role-prefix]: Variables names from within roles should use proxmox_ as a prefix. (register: linux_user_step)
roles/proxmox/tasks/admin_user.yml:2:13 Task/Handler: Create linux user for admin

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by katelovescode** [comment]
Disregard, I see where my mistake was.  For others looking at something similar, it's because I was registering a variable `linux_user_step` with no prefix.
